app/controllers/bridges/oai/oai_controller.rb

```diff
@@ -0,0 +1,268 @@
+
+          oai_response_template(xml) do
+            case verb
+              when 'GetRecord'
+                handle_get_record(xml)
+              when 'Identify'
+                handle_identify(xml)
+              when 'ListIdentifiers'
+                handle_list_identifiers(xml)
+              when 'ListMetadataFormats'
+                handle_list_metadata_formats(xml)
+              when 'ListRecords'
+                handle_list_records(xml)
+              when 'ListSets'
+                handle_list_sets(xml)
+              else
+                raise OaiError.new('Unknown verb specified', 'badVerb')
+            end
+          end
+        rescue OaiError => e
+          return render_oai_error(e)
+        end
+
+        render xml: builder.to_xml
+      end
+
+      def handle_get_record(xml)
+        task = find_record!(oai_params)
+        metadata_prefix = parse_metadata_prefix!(oai_params[:metadataPrefix])
+
+        xml.GetRecord do
+          export_record(xml, task, metadata_prefix)
+        end
+      end
+
+      def handle_identify(xml)
+        xml.Identify do
+          xml.repositoryName 'CodeHarbor'
+          xml.baseURL bridges_oai_url
+          xml.protocolVersion '2.0'
+          xml.adminEmail ADMIN_MAIL
+          tasks = Task.access_level_public.order(updated_at: :asc)
+          xml.earliestDatestamp tasks.present? ? tasks.first.updated_at.iso8601 : Time.utc(1).iso8601
+          xml.deletedRecord 'no'
+          xml.granularity 'YYYY-MM-DDThh:mm:ssZ'
+        end
+      end
+
+      def handle_list_identifiers(xml)
+        parse_metadata_prefix!(oai_params[:metadataPrefix])
+
+        xml.ListIdentifiers do
+          find_records!(oai_params).each do |task|
+            export_record_header(xml, task)
+
+            # TODO: resumptionTokens for large answers
+          end
+        end
+      end
+
+      def handle_list_metadata_formats(xml)
+        find_record!(oai_params) if params[:identifier].present?
+
+        xml.ListMetadataFormats do
+          xml.metadataFormat do
+            xml.metadataPrefix 'oai_dc'
+            xml.schema 'http://www.openarchives.org/OAI/2.0/oai_dc.xsd'
+            xml.metadataNamespace 'http://www.openarchives.org/OAI/2.0/oai_dc/'
+          end
+          xml.metadataFormat do
+            xml.metadataPrefix 'lom'
+            xml.schema 'TODO'
+            xml.metadataNamespace 'http://ltsc.ieee.org/xsd/LOM'
+          end
+        end
+      end
+
+      def handle_list_records(xml)
+        metadata_prefix = parse_metadata_prefix!(oai_params[:metadataPrefix])
+
+        xml.ListRecords do
+          find_records!(oai_params).each do |task|
+            export_record(xml, task, metadata_prefix)
+
+            # TODO: resumptionTokens for large answers
+          end
+        end
+      end
+
+      def handle_list_sets(xml)
+        xml.ListSets do
+          xml.set do
+            xml.setSpec 'task'
+            xml.setName 'Task'
```

Comment:
Having a `setName` shorter than five characters generates a warning in the validator, but I don't see an issue for that (currently).

Reply:
I would stick with this generic and short name. Since we wanted a 'global' set containing all tasks every task has this setSpec which means that harvesters display 'Task' as a tag/label for all imported tasks. At least that is what edu-sharing did.